src/components/MultiHopTrade/MultiHopTrade.tsx

```diff
@@ -62,26 +62,57 @@ export const MultiHopTrade = memo(
       exact: true,
     })
 
-    const { chainId, assetSubId, sellChainId, sellAssetSubId, sellAmountCryptoBaseUnit } =
-      match?.params || {}
+    const {
+      chainId,
+      assetSubId,
+      sellChainId,
+      sellAssetSubId,
+      sellAmountCryptoBaseUnit: paramsSellAmountCryptoBaseUnit,
+    } = match?.params || {}
 
     const sellAsset = useAppSelector(selectInputSellAsset)
     const buyAsset = useAppSelector(selectInputBuyAsset)
     const history = useHistory()
     const sellInputAmountCryptoBaseUnit = useAppSelector(selectInputSellAmountCryptoBaseUnit)
     const [isInitialized, setIsInitialized] = useState(false)
+    const [isInitialMount, setIsInitialMount] = useState(true)
 
-    const isInitialMount = useRef(true)
-
-    // Clear state *only* when mounting the main trade page (i.e not in the context of mini swapper)
     useEffect(() => {
-      // Only run this logic after the initial mount and only for the main trade page
-      if (isInitialMount.current && !isStandalone) {
-        isInitialMount.current = false
+      if (!isInitialMount || isStandalone) return
+
+      dispatch(tradeInput.actions.clear())
 
-        dispatch(tradeInput.actions.clear())
+      if (isRewritingUrl) {
+        history.push(`/trade/${buyAsset.assetId}/${sellAsset.assetId}/0`)
```

Comment:
Explicit rewrite to 0 on initial mount for this initial mount effect. It doesn't mean rehydration is broken, that's the concern of the effect below